.github/workflows/core-build.yml

```diff
@@ -7,35 +7,51 @@ on:
   workflow_call:
     inputs:
       push_image:
-        description: 'Push docker image to registry'
+        description: "Push docker image to registry"
         required: false
         type: boolean
-  
+
   workflow_dispatch:
-  
+
 jobs:
   build:
-    name: Build and test analog-core
+    name: Build analog-core
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+      - name: Restore dependencies
+        run: dotnet restore coffeecard/
+      - name: Build CoffeeCard solution
+        run: dotnet build coffeecard/ --no-restore /p:ContinuousIntegrationBuild=true
+
+  test:
```

Comment:
We end up building the solution **twice** by separating the build and test into two separate jobs. This will extend the pipeline execution time.

I suggest that we keep them in the same job